docs/02-app/01-building-your-application/06-optimizing/02-videos.mdx

```diff
@@ -0,0 +1,309 @@
+  Embed self-hosted videos using the `<video>` tag for scenarios requiring detailed control over the player's functionality and appearance. This integration method within Next.js allows for customization and control of your video content.
+- **Using Video Hosting Services (YouTube, Vimeo, etc.):**
+  For video hosting services like YouTube or Vimeo, you'll embed their iframe-based players using the `<iframe>` tag. While this method limits some control over the player, it offers ease of use and features provided by these platforms.
+
+Choose the embedding method that aligns with your application's requirements and the user experience you aim to deliver.
+
+### Embedding externally hosted videos
+
+To embed videos from platforms like YouTube or Vimeo, you can use React Suspense to stream the video component once ready.
+
+1. **Create a Server Component for Video Embedding**
+
+   The first step is to create a [Server Component](https://nextjs.org/docs/app/building-your-application/rendering/server-components) that generates the appropriate iframe for embedding the video. This component will fetch the source URL for the video and render the iframe.
+
+   ```jsx filename="app/ui/video-component.jsx"
+   export const async function VideoComponent(){
+     const src = await getVideoSrc()
+
+     return <iframe src={src} frameborder="0" allowfullscreen />
+   }
+   ```
+
+2. **Stream the Video Component Using React Suspense**
+
+   After creating the Server Component to embed the video, the next step is to [stream](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming) the component using [React Suspense](https://react.dev/reference/react/Suspense).
+
+   ```jsx filename="app/page.jsx"
+   import { Suspense } from 'react'
+   import VideoComponent from '../ui/VideoComponent.jsx'
+
+   export default function Page() {
+     return (
+       <section>
+         <Suspense fallback={<p>Loading video...</p>}>
+           <VideoComponent />
+         </Suspense>
+         {/* Other content of the page */}
+       </section>
+     )
+   }
+   ```
+
+> **Good to know**: When embedding videos from external platforms, consider the following best practices:
+>
+> - **Responsive Embedding**: Ensure the video embeds are responsive. Use CSS to make the iframe or video player adapt to different screen sizes.
+> - **Consider Data Usage**: Implement [strategies for loading videos](https://yoast.com/site-speed-tips-for-faster-video/) based on network conditions, especially for users with limited data plans.
+
+This approach results in a better user experience as it prevents the page from blocking, meaning the user can interact with the page while the video component streams in. For a more engaging and informative loading experience, consider using a loading skeleton as the fallback UI. So isntead of showing a simple loading message, you can show a skeleton that resembles the video player like this:
```

Comment:
```suggestion
This approach results in a better user experience as it prevents the page from blocking, meaning the user can interact with the page while the video component streams in. For a more engaging and informative loading experience, consider using a loading skeleton as the fallback UI. So instead of showing a simple loading message, you can show a skeleton that resembles the video player like this:
```